workers/unity/Packages/com.improbable.gdk.tools/DownloadCoreSdk.cs

```diff
@@ -136,6 +147,73 @@ private static DownloadResult Download()
             return exitCode == 0 ? DownloadResult.Success : DownloadResult.Error;
         }
 
+        private static void SetPluginsCompatibility()
+        {
+            foreach (var pluginDirectoryCompatibility in PluginsCompatibilityList)
+            {
+                var pluginPaths = AssetDatabase.FindAssets("", new[] { pluginDirectoryCompatibility.Path });
+                foreach (var pluginPath in pluginPaths)
+                {
+                    var plugin = AssetImporter.GetAtPath(AssetDatabase.GUIDToAssetPath(pluginPath)) as PluginImporter;
+                    if (plugin == null)
+                    {
+                        continue;
+                    }
+
+                    plugin.SetCompatibleWithAnyPlatform(pluginDirectoryCompatibility.AnyPlatformCompatible);
+                    plugin.SetCompatibleWithEditor(pluginDirectoryCompatibility.EditorCompatible);
+                    if (pluginDirectoryCompatibility.AnyPlatformCompatible)
+                    {
+                        foreach (var buildTarget in pluginDirectoryCompatibility.IncompatiblePlatforms)
+                        {
+                            plugin.SetExcludeFromAnyPlatform(buildTarget, true);
+                        }
+                    }
+                    else
+                    {
+                        foreach (var buildTarget in pluginDirectoryCompatibility.CompatiblePlatforms)
+                        {
+                            plugin.SetCompatibleWithPlatform(buildTarget, true);
+                        }
+                    }
+
+                    plugin.SaveAndReimport();
+                }
+            }
+        }
+
+        private class PluginDirectoryCompatibility
+        {
+            public PluginDirectoryCompatibility(string path,
+                bool anyPlatformCompatible,
+                List<BuildTarget> compatiblePlatforms,
+                List<BuildTarget> incompatiblePlatforms,
+                bool editorCompatible)
+            {
```

Comment:
Actually on second thought - if combinations of arguments can produce an invalid state - why not provide multiple constructors/static methods that restrict the argument combinations?

Reply:
Fair enough, I'll add that